src/v/cluster/controller.cc

```diff
@@ -548,59 +574,45 @@ ss::future<> controller::create_cluster() {
  * to it, and before we have started communicating with peers.
  */
 ss::future<> controller::cluster_creation_hook(cluster_discovery& discovery) {
-    auto is_cluster_founder = co_await discovery.is_cluster_founder();
-    if (!is_cluster_founder) {
-        co_return;
-    }
+    const bool controller_log_exists = _raft0->last_visible_index()
```

Comment:
Looks like `rpk_start_test` is failing, so it looks like the || size == 1 check was important in the single node case, because we may have become the leader immediately. Maybe we can just remove this entirely and rely on `is_cluster_founder`?

Reply:
For the records: checking on controller log presence was a pure race condition even in `22.2`. Bootstrap user creation may not always have worked in self-hosted envs. Ducktape environment minimizes the possibility of the race failure, so `scram_test.py::ScramBootstrapUserTest` (almost?) never failed due to that.

It's a race condition between controller and raft0 consensus. In the default unit test environment, it takes all 16 cores to initialize `drain_manager` and other sharded services before controller makes it to `cluster_creation_hook()`, and those ~25 ms are enough for consensus to update its `last_visible_index` to `0`. In ducktape though, tests are ran in containers with restricted resources, often with `--smp 1`, sharded services are started much faster, and we're in `cluster_creation_hook()` while `last_visible_index` is still the default.

`./pandaproxy_rest_fixture_rpunit -t pandaproxy_consumer_group` fails
`./pandaproxy_rest_fixture_rpunit -t pandaproxy_consumer_group -- -c 1` passes
`./pandaproxy_rest_fixture_rpunit -t pandaproxy_consumer_group -- --overprovisioned` passes too

I'm removing the check for nonempty controller log altogether.